pallets/account-linker/src/tests.rs

```diff
@@ -45,20 +78,134 @@ fn it_works_for_default_value() {
 	});
 }
 
-// account: T::AccountId,
-// index: u32,
-// block_number: T::BlockNumber,
-// r: [u8; 32],
-// s: [u8; 32],
-// v: u8,
-
-// #[test]
-// fn correct_error_for_none_value() {
-// 	new_test_ext().execute_with(|| {
-// 		// Ensure the expected error is thrown when no value is present.
-// 		assert_noop!(
-// 			TemplateModule::cause_error(Origin::signed(1)),
-// 			Error::<Test>::NoneValue
-// 		);
-// 	});
-// }
+
+#[test]
+fn test_invalid_block_number() {
+	new_test_ext().execute_with(|| {
+
+		let account: u64 = 5;
+		let block_number: u64 = 0;
+
+		let mut gen = Random{};
+		let key_pair = gen.generate().unwrap();
+
+		let msg = generate_msg(account, block_number);
+		let sig = generate_sig(&key_pair, &msg);
+		let (r, s, v) = generate_rsv(&sig);
+
+		let result = AccountLinker::link(Origin::signed(1), account, 0, block_number, r, s, v);
+		assert_eq!(result.is_err(), true);
+		assert_eq!(result.err(), Some(DispatchError::from(AccountLinkerError::LinkRequestExpired)));
+	});
+}
+
+#[test]
+fn test_insert_eth_address() {
+	new_test_ext().execute_with(|| {
+
+		let account: u64 = 5;
+		let block_number: u64 = 99999;
+
+
+		let mut gen = Random{};
+		for i in 0..(MAX_ETH_LINKS) {
+
+			let key_pair = gen.generate().unwrap();
+
+			let msg = generate_msg(account, block_number + i as u64);
+			let sig = generate_sig(&key_pair, &msg);
+
+			let (r, s, v) = generate_rsv(&sig);
+
+			let _ = AccountLinker::link(Origin::signed(1),
+										account,
+										i as u32,
+										block_number + i as u64,
+										r,
+										s,
+										v);
+			assert_eq!(AccountLinker::eth_addresses(account).len(), i+1);
+
+			let inserted_addr = AccountLinker::eth_addresses(account)[i];
+			let recovered_addr = util::addr_from_sig(msg.into(), sig).ok().unwrap();
```

Comment:
Here we are not testing `util::addr_from_sig`. Therefore we need to compare the inserted_addr with original intentional addr
Besides, please compare the whole vec instead of just one member of the vec

Reply:
see file https://github.com/litentry/litentry-node/blob/develop/pallets/account-linker/src/util.rs.

You tested !